### PR TITLE
MoltenVKShaderConverter updates

### PR DESCRIPTION
**Pull request by billhollings**
- `mvk::getShaderOutputs()` in `SPRIVReflection.h` support flattening nested structures.
- `MoltenVKShaderConverter` tool support loading tessellation shader files.
- `MoltenVKShaderConverter` tool update to MSL 2.4 by default.
- Remove use of deprecated `MTLCreateSystemDefaultDevice()`.
- Update to latest version of SPIRV-Cross.

### CODE REVIEW

File: MoltenVKShaderConverter/MoltenVKShaderConverterTool/OSSupport.mm [hunk 1]
```diff
@@ -100,12 +100,18 @@
...
+
 		MTLCompileOptions* mtlCompileOptions  = [[MTLCompileOptions new] autorelease];
 		mtlCompileOptions.languageVersion = mslVerEnum;
 		NSError* err = nil;
-		id<MTLLibrary> mtlLib = [[MTLCreateSystemDefaultDevice() newLibraryWithSource: @(mslSourceCode.c_str())
-																			  options: mtlCompileOptions
-																				error: &err] autorelease];
+		id<MTLLibrary> mtlLib = [[mtlDevs[0] newLibraryWithSource: @(mslSourceCode.c_str())
+														  options: mtlCompileOptions
+															error: &err] autorelease];
```
Comment (cdavis5e) [inline MoltenVKShaderConverter/MoltenVKShaderConverterTool/OSSupport.mm:114]: This change does not match the description in the commit message, which confused me.

File: MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVReflection.h [hunk 2]
```diff
@@ -173,6 +173,42 @@ namespace mvk {
...
+	template<typename Vo>
+	static inline uint32_t getShaderOutputStructMembers(const SPIRV_CROSS_NAMESPACE::CompilerReflection& reflect, Vo& outputs,
+														const SPIRV_CROSS_NAMESPACE::SPIRType* structType, spv::StorageClass storage,
+														bool patch, uint32_t loc) {
+		bool isUsed = true;
+		auto biType = spv::BuiltInMax;
+		size_t mbrCnt = structType->member_types.size();
+		for (uint32_t mbrIdx = 0; mbrIdx < mbrCnt; mbrIdx++) {
+			// Each member may have a location decoration. If not, each member
+			// gets an incrementing location based the base location for the struct.
```
Comment (cdavis5e) [inline MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVReflection.h:187]:
```suggestion
			// gets an incrementing location based on the base location for the struct.
```
I know that was there in the original, but this seems like a good time to fix the typo.

**Review by cdavis5e — APPROVED** [review]
This actually looks a lot better now. Thanks for cleaning up my really redundant code.

File: MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVReflection.h (hunk 2 above)
Comment (billhollings) [reply MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVReflection.h]: Done.

File: MoltenVKShaderConverter/MoltenVKShaderConverterTool/OSSupport.mm (hunk 1 above)
Comment (billhollings) [reply MoltenVKShaderConverter/MoltenVKShaderConverterTool/OSSupport.mm:114]: Oh geez. I got the commit message description backwards. Fixed now.

Comment (billhollings) [reply MoltenVKShaderConverter/MoltenVKShaderConverterTool/OSSupport.mm:114]: Oh, geez. I got the commit message description wrong. Fixed now.

**Pull request merged**